src/payment/strategies/quadpay/quadpay-payment-strategy.ts

```diff
@@ -38,15 +38,16 @@ export default class QuadpayPaymentStrategy implements PaymentStrategy {
         let nonce: string;
         const { methodId } = payment;
         const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId, options));
-        const { clientToken = '' } = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+        const { clientToken = '', initializationData = '' } = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+        const redirectUrl = initializationData?.redirectUrl;
 
         try {
             ({ id: nonce } = JSON.parse(clientToken));
         } catch (error) {
             throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
```

Comment:
Minor one: Do we need a new error type here?

Reply:
@animesh1987, I think that error type is the best choice, if something went wrong with the `JSON.parse` means that the payment method data was unavailable or it was not correct, or which one do you think is a better option?